win32ss/gdi/ntgdi/freetype.c

```diff
@@ -2994,38 +2993,103 @@ static unsigned int get_bezier_glyph_outline(FT_Outline *outline, unsigned int b
 static INT
 IntRequestFontSize(PDC dc, FT_Face face, LONG Width, LONG Height)
 {
-    FT_Size_RequestRec  req;
+    FT_Error error;
+    ULONG TargetHeight, TargetWidth, RealHeight;
 
-    if (Width < 0)
-        Width = -Width;
+    ASSERT(face != NULL);
+    ASSERT(face->height > 0);
+    ASSERT(face->units_per_EM > 0);
+
+    if (Height == 0)
+    {
+        Height = dc->ppdev->devinfo.lfDefaultFont.lfHeight;
+    }
 
     if (Height < 0)
     {
-        Height = -Height;
+        /* case (A): lfHeight is negative: the character height */
+        Height = -Height << 6;
```

Comment:
What's the meaning of the magic 64?

Reply:
FreeType often uses 26.6 (26+6 == 32 bits) fixed point numbers. See FreeType documents.